src/modal/modal.js

```diff
@@ -254,12 +254,12 @@ class Modal extends React.Component<ModalPropsT, ModalStateT> {
               {...sharedProps}
               {...rootProps}
             >
-              <Backdrop
-                onClick={this.onBackdropClick}
-                {...sharedProps}
-                {...backdropProps}
-              />
               <DialogContainer {...sharedProps} {...dialogContainerProps}>
+                <Backdrop
```

Comment:
moving this element is a bit dangerous considering if anyone overriding the modal relies on it, but i suppose it's fixing a bug therefore justified